### PR TITLE
Changes required for desktop wallet backport (+ other things)

### PR DESCRIPTION
**Pull request by pablen**
- Some modifications to existing logic so that this module can be reused in desktop wallet project
- Add some missing logic not currently present in mobile version (like offline warnings)
- Removed all legacy references to "MTN"
- Initial work to support multiple chains
- Refactored config shape

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+save-exact = true
```
Comment (gabmontes) [inline .npmrc]: We should not do this in a lib unless we have a very good reason i.e. newer versions of a specific package break things. In that case we should only fix those.

File: src/hocs/withPortState.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        blockTimestamp: tx.meta.blockTimestamp,
+        dailyMintable: tx.meta.dailyMintable,
+        burnSequence: tx.meta.burnSequence,
+        currentTick: tx.meta.currentTick,
+        originChain: tx.originChain,
+        value: tx.meta.value,
+        from: tx.receipt.from,
+        fee: tx.meta.fee
+
+        // TODO: define these parameters
```
Comment (gabmontes) [inline src/hocs/withPortState.js]: Should we remove this?

File: .npmrc (hunk 1 above)
Comment (pablen) [reply .npmrc]: 👍

File: src/hocs/withPortState.js (hunk 2 above)
Comment (pablen) [reply src/hocs/withPortState.js]: Yes, we should.

**Review by gabmontes — APPROVED** [review]

**Pull request merged**